posts/inside-rust/2025-01-17-this-development-cycle-in-cargo-1.85.md

```diff
@@ -0,0 +1,388 @@
+Purely additive changes to a Summary are fine,
+like adding `package.rust-version`.
+We can extend existing fields or change the interpretation of them but the experience isn't ideal for old versions of Cargo.
+
+Back when Cargo 1.60 was released,
+users on older versions of Cargo started to see errors like
+```
+error: failed to select a version for the requirement `libgit2-sys = "^0.13.3"`
+candidate versions found which didn't match: 0.13.2+1.4.2, 0.13.1+1.4.2, 0.13.0+1.4.1, ...
+location searched: crates.io index
+required by package `git2 v0.14.3`
+    ... which satisfies dependency `git2 = "^0.14"` (locked to 0.14.3) of package `repor v0.1.0 (/home/epage/src/personal/repro)`
+```
+When you go on crates.io, version `0.13.3` exists.
+The problem is `0.13.3` used the new [weak dependency feature syntax](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features) (`dep?/feature`) which older Cargos do not understand.
+This error message is a poor user experience for weak dependency features and any future extensions of the Summary schema.
+We've been tracking this usability problem in [#10623](https://github.com/rust-lang/cargo/issues/10623).
+
+[epage](https://github.com/epage) did some recent work (
+[#14897](https://github.com/rust-lang/cargo/pull/14897),
+[#14921](https://github.com/rust-lang/cargo/pull/14921),
+[#14923](https://github.com/rust-lang/cargo/pull/14923),
+[#14927](https://github.com/rust-lang/cargo/pull/14927)
+) so Cargo can extract as much information as it can from unsupported Summaries, without losing this information through caching, and report it to the user.
+
+The following output shows three different ways we might report an unsupported Summary, depending on how much context we have:
+```
+[UPDATING] `dummy-registry` index
+[ERROR] failed to select a version for the requirement `foo = "^0.1.1"`
+  version 0.1.3 requires cargo 1.2345
+  version 0.1.4 requires a Cargo version that supports index version 1000000000
+  version 0.1.5's index entry is invalid
+location searched: `dummy-registry` index (which is replacing registry `crates-io`)
+required by package `a v0.5.0 ([ROOT]/foo)`
+```
+We also report yanked versions now:
+```
+[UPDATING] `dummy-registry` index
+[ERROR] failed to select a version for the requirement `baz = "=0.0.2"`
+  version 0.0.2 is yanked
+location searched: `dummy-registry` index (which is replacing registry `crates-io`)
+required by package `bar v0.0.1`
+    ... which satisfies dependency `bar = "*"` of package `foo v0.0.1 ([ROOT]/foo)`
+```
+
+With the improved error reporting,
+we are more free to design Index related features.
+
+## Design discussions
+
+### Project goals
+
+Cargo-related goals that have been proposed for 2025h1 include
+- [build-std](https://rust-lang.github.io/rust-project-goals/2025h1/build-std.html): led by [davidtwco](https://github.com/davidtwco)
```

Comment:
```suggestion
- [build-std](https://rust-lang.github.io/rust-project-goals/2025h1/build-std.html): led by [AdamGemmell](https://github.com/AdamGemmell) and the Rust team at Arm
```
While I'll be involved and write the goal updates and things like that, I won't personally be doing the bulk of the work, so don't want to take the credit here for leading the goal as-such, Adam will be doing the work here so I think he should be highlighted